store/src/main/java/org/apache/rocketmq/store/MappedFileQueue.java

```diff
@@ -247,11 +247,18 @@ public boolean doLoad(List<File> files) {
         // ascending order
         files.sort(Comparator.comparing(File::getName));
 
-        for (File file : files) {
+        for (int i = 0; i < files.size(); i++) {
+            File file = files.get(i);
             if (file.isDirectory()) {
                 continue;
             }
 
+            if (file.length() == 0 && i == files.size() - 1) {
```

Comment:
We should investigate why there are files with a length of 0. Is it caused by pre-allocation or file switching?
If any file has a length of zero, it can be considered as damaged, Not only the last one.

Reply:
The file is empty because broker exits after file created and before mmap, new RandomAccessFile will create an empty file.

![image](https://github.com/apache/rocketmq/assets/14816818/064d1053-eeea-496e-a5c7-da0fe7a722f6)